src/main/steps/applicant/statement-of-service/summary/StatementOfServicePostController.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { getPartyDetails, mapDataInSession } from '../../../../../main/steps/tasklistresponse/utils';
+import {
+  APPLICANT_STATEMENT_OF_SERVICE_NEXT,
+  APPLICANT_STATEMENT_OF_SERVICE_SUMMARY,
+} from '../../../../../main/steps/urls';
+import { CosApiClient } from '../../../../app/case/CosApiClient';
+import { CaseEvent } from '../../../../app/case/definition';
+import { AppRequest } from '../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../app/controller/PostController';
+import { FormFields, FormFieldsFn } from '../../../../app/form/Form';
+import { applyParms } from '../../../../steps/common/url-parser';
+import { prepateStatementOfServiceRequest } from '../choose-parties/StatementOfServiceMapper';
+
+@autobind
+export default class StatementOfServicePostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const { sosConsent } = req.body;
+    if (!sosConsent || sosConsent === '') {
```

Comment:
Why is the error validation done manually. use framework form error for validation and redirect to same page if error occurs

Reply:
Updated